jobqueue.go

```diff
@@ -110,11 +149,35 @@ func (s *Stack) run() {
 				j := s.stack.pop()
 				j.notify <- nil
 			}
+
+			if s.closing && s.busy == 0 && s.stack.empty() {
```

Comment:
can s.busy be 0 and stack.empty() return false?
As far as I see from line 145-151 it can not happen.

Reply:
If I understand the question right, then yes I think it can. The preceeding code only acknowledges that a single job is done, and if there is any waiting, then allows a single one to run, but there can be many more running and waiting.